src/raster/rasterband.rs

```diff
@@ -485,6 +487,13 @@ impl<'a> RasterBand<'a> {
         Some(ColorTable::from_c_color_table(c_color_table))
     }
 
+    /// Set the color table for this band.
+    ///
+    /// See [`ColorTable`] for usage example.
+    pub fn set_color_table(&mut self, colors: &ColorTable) {
+        unsafe { GDALSetRasterColorTable(self.c_rasterband, colors.c_color_table) };
```

Comment:
NB: Per the GDAL docs, `GDALSetRasterColorTable` copies the entries from the given `ColorTable`.